exchange-v2/contracts/ExchangeV2Core.sol

```diff
@@ -98,8 +163,72 @@ abstract contract ExchangeV2Core is Initializable, OwnableUpgradeable, AssetMatc
 
     function validateFull(LibOrder.Order memory order, bytes memory signature) internal view {
         LibOrder.validate(order);
-        validate(order, signature);
+        //no need to validate signature of an onchain order
+        if (!checkOrderExistance(LibOrder.hashKey(order))){
+            validate(order, signature);
+        } 
+    }
+
+    /// @dev Checks order for existance onchain
+    function checkOrderExistance(bytes32 orderKeyHash) public view returns(bool){
+        if(onChainOrders[orderKeyHash].maker != address(0)){
```

Comment:
let's check code style. we use spaces before paretheses. pls, check style to conform to our